src/detail/unique.hpp

```diff
@@ -0,0 +1,32 @@
+#pragma once
+#include <concepts>
+#include <utility>
+
+namespace dibs::detail {
+template <typename T>
+concept comparable = requires(T const& a, T const& b) {
+	{ a == b } -> std::convertible_to<bool>;
+};
+
+template <comparable T, typename Deleter = void>
+struct Unique {
+	using type = T;
+	using deleter_t = Deleter;
+
+	T t;
+
+	constexpr Unique(T t = T{}) noexcept : t(std::move(t)) {}
```

Comment:
Should we add a check to ensure `T` is default constructable or will `T` ever only be an aggregate?

Reply:
Good idea!

Reply:
The class template could take a second non type template parameter: `Unique<T, T Null = T{}>`, that way it could be used with some weird special types too.

Reply:
Would you like to try to refactor that (in a separate PR)?

Reply:
> The class template could take another (non type) template parameter: `Unique<T, Deleter = void, T Null = T{}>`, that way it could be used with some weird special types too.

Can you give some examples?